packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -997,3 +1073,171 @@ capacity provider. The weight value is taken into consideration after the base v
    */
   readonly weight?: number;
 }
+
+/**
+ * An ECS Capacity Provider
+ */
+export interface ICapacityProvider {
+  /**
+   * The name for the capacity provider.
+   *
+   * @default CloudFormation-generated name
+   */
+  readonly capacityProviderName: string;
+}
+
+/**
+ * A class used solely for generating the singleton objects that follow
+ */
+class NamedCapacityProvider implements ICapacityProvider {
+  public capacityProviderName: string;
+  constructor(name: string) {
+    this.capacityProviderName = name;
+  }
+}
+
+/**
+ * Fargate capacity provider
+ */
+export const FargateCapacityProvider = new NamedCapacityProvider('FARGATE');
+
+/**
+ * Fargate Spot capacity provider
+ */
+export const FargateSpotCapacityProvider = new NamedCapacityProvider('FARGATE_SPOT');
+
+/**
+ * The options for creating an EC2 Capacity Provider.
+ */
+export interface EC2CapacityProviderProps extends AddAutoScalingGroupCapacityOptions {
+  /**
+   * The name for the capacity provider.
+   *
+   * @default CloudFormation-generated name
+   */
+  readonly capacityProviderName?: string;
+
+  /**
+   * The autoscaling group to add as a Capacity Provider.
+   * [disable-awslint:ref-via-interface] is needed to enable managed termination
+   * protection.
+   */
+  readonly autoScalingGroup: autoscaling.AutoScalingGroup;
```

Comment:
I have never understood why `IAutoScalingGroup` is a thing, because there's only one implementation of the abstract type, and I can't think of a situation in which we would ever have more than one. Also, per my comment, I wasn't able to enable managed protection on the abstract class, since it's, well, abstract. I'd have to cast it in the implementation anyway.

Reply:
This is to ensure that you can import an ASG defined in a different stack, I believe.

Would it be possible to add the managed protection methods to the abstract class/interface?